langchain4j-open-ai-official/src/main/java/dev/langchain4j/model/openaiofficial/OpenAiOfficialExceptionMapper.java

```diff
@@ -0,0 +1,29 @@
+package dev.langchain4j.model.openaiofficial;
+
+import com.openai.errors.OpenAIIoException;
+import dev.langchain4j.Internal;
+import dev.langchain4j.exception.HttpException;
+import dev.langchain4j.exception.LangChain4jException;
+import dev.langchain4j.exception.TimeoutException;
+import dev.langchain4j.internal.ExceptionMapper;
+import java.io.InterruptedIOException;
+
+@Internal
+class OpenAiOfficialExceptionMapper extends ExceptionMapper.DefaultExceptionMapper {
+
+    static final OpenAiOfficialExceptionMapper INSTANCE = new OpenAiOfficialExceptionMapper();
+
+    @Override
+    public RuntimeException mapException(Throwable t) {
+        if (t instanceof com.openai.errors.OpenAIServiceException httpResponseException) {
+            final var statusCode = httpResponseException.statusCode();
+            return mapHttpStatusCode(new HttpException(statusCode, httpResponseException), statusCode);
```

Comment:
```suggestion
            return mapHttpStatusCode(httpResponseException, statusCode);
```
why adding an extra layer of exceptions?

Reply:
Once simplified, please remove new ctors in `HttpException` which are not required any more